server/SessionService.java

```diff
@@ -62,6 +62,12 @@ void register(TransactionService transactionSvc) {
             if (isOpen.get()) {
                 transactionServices.add(transactionSvc);
                 cancelIdleTimeout();
+                if (LOG.isDebugEnabled()) {
```

Comment:
We protect the logging in an `if` when the logging can be expensive: computing the size of concurrent hash maps/sets could induce expensive locking and block other operations - the contract does not guarantee this is a fast operation.